public/views/jQuery.js

```diff
@@ -0,0 +1,141 @@
+$(document).ready(function() {
+
+  scorecard = new BowlingScorecard();
+
+  var outputCounter = 0;
+  var numberChecker = 0;
+  printTotal();
+
+  $("#button0").on('click', function() {
+    if (isGameOver(outputCounter) == false) {
+      $("#button0").html(scorecard.roll(0));
+      printButton(0);
+      printTotal();
+    }
+  });
+  $("#button1").on('click', function() {
+    if (isGameOver(outputCounter) == false) {
+      $("#button1").html(scorecard.roll(1));
+      printButton(1);
+      printTotal();
+    }
+  });
+  $("#button2").on('click', function() {
+    if (isGameOver(outputCounter) == false) {
+      $("#button2").html(scorecard.roll(2));
+      printButton(2);
+      printTotal();
+    }
+  });
+  $("#button3").on('click', function() {
+    if (isGameOver(outputCounter) == false) {
+      $("#button3").html(scorecard.roll(3));
+      printButton(3);
+      printTotal();
+    }
+  });
+  $("#button4").on('click', function() {
+    if (isGameOver(outputCounter) == false) {
+      $("#button4").html(scorecard.roll(4));
+      printButton(4);
+      printTotal();
+    }
+  });
+  $("#button5").on('click', function() {
+    if (isGameOver(outputCounter) == false) {
+      $("#button5").html(scorecard.roll(5));
+      printButton(5);
+      printTotal();
+    }
+  });
+  $("#button6").on('click', function() {
+    if (isGameOver(outputCounter) == false) {
+      $("#button6").html(scorecard.roll(6));
+      printButton(6);
+      printTotal();
+    }
+  });
+  $("#button7").on('click', function() {
+    if (isGameOver(outputCounter) == false) {
+      $("#button7").html(scorecard.roll(7));
+      printButton(7);
+      printTotal();
+    }
+  });
+  $("#button8").on('click', function() {
+    if (isGameOver(outputCounter) == false) {
+      $("#button8").html(scorecard.roll(8));
+      printButton(8);
+      printTotal();
+    }
+  });
+  $("#button9").on('click', function() {
+    if (isGameOver(outputCounter) == false) {
+      $("#button9").html(scorecard.roll(9));
+      printButton(9);
+      printTotal();
+    }
+  });
+  $("#button10").on('click', function() {
+    if (isGameOver(outputCounter) == false) {
+      $("#button10").html(scorecard.roll(10));
+      printButton(10);
+      printTotal();
+    }
+  });
+
+  function printButton(number) {
+    outputCounter += 1;
```

Comment:
I'm not sure I totally understand what this is doing, but I think this is trying to work out which frame to display the score on. If that's the case, a better separation of concerns would be:

```
var currentFrame = scorecard.getCurrentFrame();
$('#' + currentFrame).text(number);
```

i.e. your view logic doesn't need to keep track any outputCounter - it just asks the scorecard 'what's the current frame?', and when the score card tells it, it understand that it needs to apply the last roll to the view element representing that frame.

Event better though, if you follow an OO approach and you treat each of the individual frames as objects, you can get the UI to listen for changes in the frames; 

```
// Define a frame class
function Frame() {
   var score = 0;
   var listener = null;

   function setScore(roll) {
      score = roll;
      if (listener !== null) {
         listener(score);
      }
   }

   function getScore() {
      return score;
   }

   function addListener(scoreListener) {
      listener = scoreListener;
   }
   
   return {
      setScore: setScore,
      addListener: addListener
   }
}

...
// In your view logic
var buttons = $('.buttons);
for (var index = 0; index < buttons.length; index++) {
   var frame = scoreCard.getFrame(index);
   frame.addListener(function(score) {
      updateScore(buttons[index], score);
   });
}
```

This last example is pretty advanced and may be difficult to follow, but it's a good place to end up if you can :)